### ISSUE TITLE
[Tracking] linux 5.2 dkms status

### ISSUE THREAD

**Issue by Hoshpak** [void-linux/void-packages#12959]
These dkms modules are important and their compatibility with a kernel version is relevant for the decision to move the default kernel version:

- [x] nvidia
- [x] nvidia390
- [x] nvidia340 (compilation and runtime tested)
- [ ] zfs (will probably have to wait for the next release)
- [x] virtualbox-ose (compilation and runtime tested)

Less important modules:
- [x] rtl8812au-dkms
- [x] virtualbox-ose-guest
- [ ] exfat-dkms (broken)

If you are using any of these modules, please consider contributing to this issue by installing `linux5.2` and `linux5.2-headers` and report if everything still works for you.

**Comment by Duncaen** [comment]
`nvidia-430.14_2` works here.

**Comment by abenson** [comment]
Looks like nvidia390 needs another patch, but I didn't get a chance to look into it. I'll check tonight.

**Comment by rapha8l** [comment]
linux5.2-5.2.0_1: configuring ...
Executing post-install kernel hook: 10-dkms ...
Available DKMS module: acpi_call-1.2.0.
Available DKMS module: nvidia-430.14.
Available DKMS module: virtualbox-ose-6.0.8.
Available DKMS module: virtualbox-ose-guest-6.0.8.
Available DKMS module: zfs-0.8.1.
Building DKMS module: acpi_call-1.2.0... done.
Installing DKMS module: acpi_call-1.2.0... done.
Building DKMS module: nvidia-430.14... done.
Installing DKMS module: nvidia-430.14... done.
Building DKMS module: virtualbox-ose-6.0.8... done.
Installing DKMS module: virtualbox-ose-6.0.8... done.
Building DKMS module: virtualbox-ose-guest-6.0.8... FAILED!

ZFS dkms works on another machine

for virtualbox : 
make.log added
[make.log](https://github.com/void-linux/void-packages/files/3386356/make.log)

thanks

**Comment by renatoaguiar** [comment]
rtl8812au-dkms is failing.

**Comment by renatoaguiar** [comment]
rtl8812au-dkms being fixed by https://github.com/void-linux/void-packages/pull/13366

**Comment by Hoshpak** [comment]
virtualbox-ose-guest was fixed by the update to 6.0.10

**Comment by Hoshpak** [comment]
nvidia390 should be fixed by 99148f5768f87f16de3d665144e100143496eed0 , please test.

**Comment by yilmzfurkan** [comment]
@Hoshpak successfully upgraded to linux5.2 with nvidia390. Void Linux x86_64

**Comment by steinex** [comment]
There is a pretty bad amdgpu regression in 5.2 not yet fixed in stable: https://bugzilla.kernel.org/show_bug.cgi?id=204181

**Comment by maop** [comment]
```
linux5.2-5.2.4_1: configuring ...
Executing post-install kernel hook: 10-dkms ...
Available DKMS module: nvidia-390.116.
Available DKMS module: rtl8192eu-1.0.
Building DKMS module: nvidia-390.116... done.
Installing DKMS module: nvidia-390.116... done.
Building DKMS module: rtl8192eu-1.0... done.
Installing DKMS module: rtl8192eu-1.0... done.
```
nvidia390 and rtl8192eu compiles OK

**Comment by codingHahn** [comment]
Any news on this?

**Comment by Hoshpak** [comment]
waiting for a new zfs release, waiting for amdgpu to get fixed. Since there's not much movement in either direction, my buest guess is that 5.2 won't become the default kernel, 5.3 or 5.4 might have a chance later on.

**Comment by deepcube** [comment]
exfat-dkms fails:
`
DKMS make.log for exfat-1.2.8 for kernel 5.2.9_1 (x86_64)
Fri 23 Aug 2019 10:08:20 AM PDT
make -C /lib/modules/5.2.9_1/build M=/var/lib/dkms/exfat/1.2.8/build modules
make[1]: Entering directory '/usr/src/kernel-headers-5.2.9_1'
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_core.o
/var/lib/dkms/exfat/1.2.8/build/exfat_core.c: In function 'fs_error':
/var/lib/dkms/exfat/1.2.8/build/exfat_core.c:1760:64: error: 'MS_RDONLY' undeclared (first use in this function); did you mean 'IS_RDONLY'?
 1760 |  else if ((opts->errors == EXFAT_ERRORS_RO) && !(sb->s_flags & MS_RDONLY)) {
      |                                                                ^~~~~~~~~
      |                                                                IS_RDONLY
/var/lib/dkms/exfat/1.2.8/build/exfat_core.c:1760:64: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [scripts/Makefile.build:279: /var/lib/dkms/exfat/1.2.8/build/exfat_core.o] Error 1
make[1]: *** [Makefile:1597: _module_/var/lib/dkms/exfat/1.2.8/build] Error 2
make[1]: Leaving directory '/usr/src/kernel-headers-5.2.9_1'
make: *** [Makefile:35: all] Error 2
`

While figuring out how to fix this I saw that it was already reported on the 5.0 ticket.
https://github.com/void-linux/void-packages/issues/9699

Adding the patch mentioned there works.
https://github.com/dorimanx/exfat-nofuse/issues/145

**Comment by Hoshpak** [comment]
After applying the patch, another error occurs in another part of the code:

```
DKMS make.log for exfat-1.2.8 for kernel 5.2.5_1 (x86_64)
Sa 24. Aug 15:02:50 CEST 2019
make -C /lib/modules/5.2.5_1/build M=/var/lib/dkms/exfat/1.2.8/build modules
make[1]: Verzeichnis „/usr/src/kernel-headers-5.2.5_1“ wird betreten
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_core.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_super.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_api.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_blkdev.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_cache.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_data.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_bitmap.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_nls.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_oal.o
/var/lib/dkms/exfat/1.2.8/build/exfat_oal.c: In function 'tm_current':
/var/lib/dkms/exfat/1.2.8/build/exfat_oal.c:137:2: error: implicit declaration of function 'ktime_get_real_ts'; did you mean 'ktime_get_real_ns'? [-Werror=implicit-function-declaration]
  137 |  ktime_get_real_ts(&ts);
      |  ^~~~~~~~~~~~~~~~~
      |  ktime_get_real_ns
cc1: some warnings being treated as errors
```
I'm not interested in this module at all, I'm just leaving it here for people who want to look into patching it.

Since this is purely a tracking issue, it would be better to create a separate issue for the respective problems that arise. This way it gets more visibility and won't be lost like in the 5.0 discussion.

**Comment by yilmzfurkan** [comment]
I am back from vacation and I have problems with nvidia390 in linux5.2.13_1 building it fails. "Building DKMS module: nvidia-390.116... FAILED!" The latest version which I know it works is linux5.2.10_1

**Comment by Hoshpak** [comment]
This is most likely caused by the gcc update and not by the kernel version. The nvidia driver checks if the gcc version used to compile the kernel matches with the gcc version used to compile the module and refuses to build otherwise. You can override this by setting `IGNORE_CC_MISMATCH=1` in the environment.

I'm closing this issue since 5.2 is now the default kernel and the dkms status is thus no longer relevant.

**Comment by yilmzfurkan** [comment]
@Hoshpak okay, thanks. Just was going to edit it that "sudo xbps-reconfigure -f nvidia390-dkms" fixed it.
